pylint/testutils/configuration_test.py

```diff
@@ -0,0 +1,101 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+"""Utility function for configuration testing."""
+import copy
+import json
+import logging
+import unittest
+from pathlib import Path
+from typing import Any, Dict, Tuple, Union
+from unittest.mock import Mock
+
+from pylint.lint import Run
+
+USER_SPECIFIC = str(Path(__file__).parent.parent.parent)
+
+
+def get_expected_or_default(pyproject_toml_path: str, suffix: str, default: Any) -> str:
+    def get_path_according_to_suffix() -> Path:
+        path = Path(pyproject_toml_path)
+        return path.parent / f"{path.stem}.{suffix}"
+
+    expected = default
+    expected_result_path = get_path_according_to_suffix()
+    if expected_result_path.exists():
+        with open(expected_result_path, encoding="utf8") as f:
+            expected = f.read()
+        logging.info("%s exists.", expected_result_path)
+    else:
+        logging.info("%s not found, using '%s'.", expected_result_path, default)
+    return expected
+
+
+EXPECTED_CONF_APPEND_KEY = "functional_append"
+EXPECTED_CONF_REMOVE_KEY = "functional_remove"
+EXPECTED_CONF_SPECIAL_KEYS = [EXPECTED_CONF_APPEND_KEY, EXPECTED_CONF_REMOVE_KEY]
+
+
+def get_expected_configuration(
+    configuration_path: str, default_configuration: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Get the expected parsed configuration of a configuration functional test"""
+    result = copy.deepcopy(default_configuration)
+    config_as_json = get_expected_or_default(
+        configuration_path, suffix="result.json", default="{}"
+    )
+    to_override = json.loads(config_as_json)
+    for key, value in to_override.items():
+        if key not in EXPECTED_CONF_SPECIAL_KEYS:
+            result[key] = value
+    if EXPECTED_CONF_APPEND_KEY in to_override:
```

Comment:
Can't this be done in the `for` loop with an `elif`?
Similar for the `if` on L54.

Reply:
2eaad656d8c919dc9b007e1886e1d53d71fd69f4